pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -650,29 +650,18 @@ protected CompletableFuture<Void> internalGrantPermissionOnSubscriptionAsync(Str
     protected CompletableFuture<Void> internalRevokePermissionsOnNamespaceAsync(String role) {
         return validateNamespaceOperationAsync(namespaceName, NamespaceOperation.REVOKE_PERMISSION)
                 .thenAccept(__ -> checkNotNull(role, "Role should not be null"))
-                .thenCompose(__ -> validatePoliciesReadOnlyAccessAsync())
-                .thenCompose(__ -> updatePoliciesAsync(namespaceName, policies -> {
-                    policies.auth_policies.getNamespaceAuthentication().remove(role);
-                    return policies;
-                }));
+                .thenCompose(__ -> getAuthorizationService().revokePermissionAsync(namespaceName, role));
     }
 
     protected CompletableFuture<Void> internalRevokePermissionsOnSubscriptionAsync(String subscriptionName,
                                                                                   String role) {
-        AuthorizationService authService = pulsar().getBrokerService().getAuthorizationService();
-        if (null != authService) {
```

Comment:
Why are we removing this check? There are still 2 other references in this file where we fail if the authorization service is null. Seems like we should maintain a consistent design and either keep it or remove it completely. I'd prefer to make the failure clearer and fail right explicitly instead of due to an NPE.

Reply:
I think `authService ` should not be null. 

> There are still 2 other references in this file 

I have removed them

Reply:
You are right, thanks for the explanation.